File: APT1030_Coursework/Level5/ride.cpp
#include <iostream>
using namespace std;

// Procedural Function: Focuses on the action (calculating)
double calculateFare(double distance) {
    double baseFare = 200.0;
    double costPerKm = 50.0;
    return baseFare + (distance * costPerKm);
}

int main() {
    double distance;
    cout << "--- Nairobi Ride-Hailing (Procedural C++) ---" << endl;
    cout << "Enter distance in KM: ";
    cin >> distance;

    double total = calculateFare(distance);
    cout << "Total Fare: " << total << " KES" << endl;

    return 0;
}

File: APT1030_Coursework/Level5/ride.py
# Functional Approach: No classes, no side effects
base_fare = 200
cost_per_km = 50

# Lambda Function: (Input Distance 'd') -> Result
calc_fare = lambda d: base_fare + (d * cost_per_km)

def run_functional_engine():
    print("--- Nairobi Ride-Hailing (Functional Python) ---")
    try:
        dist = float(input("Enter distance in KM: "))
        # We "apply" the lambda function to the input
        total = calc_fare(dist)
        print(f"Total Fare: {total:,.2f} KES")
    except ValueError:
        print("Invalid input. Please enter a number.")

if __name__ == "__main__":
    run_functional_engine()